src/vision/calib/Calibration.cpp

```diff
@@ -0,0 +1,210 @@
+#include "MonoCalibrator.hpp"
+#include "wfov_camera_msgs/WFOVImage.h"
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <string>
+#include <vector>
+
+using namespace cv_bridge;
+
+Settings s;
+
+bool calibFinished = false;
+vector<vector<Point2f>> imagePoints;
+Mat cameraMatrix, distCoeffs;
+Size imageSize;
+int mode = CAPTURING;
+clock_t prevTimestamp = 0;
+const Scalar RED(0, 0, 255), GREEN(0, 255, 0);
+const char ESC_KEY = 27;
+
+void imageCallback(const wfov_camera_msgs::WFOVImage::ConstPtr& msg)
+{
+    ROS_DEBUG_STREAM("Callback, Mode: " << mode);
+    Mat view = toCvShare(msg->image, msg,
+                         sensor_msgs::image_encodings::BGR8)->image;
+
+    if (mode == CAPTURING && imagePoints.size() >= (size_t)s.nrFrames)
+    {
+        ROS_DEBUG_STREAM("Calibrating and saving during capture");
+        if (runCalibrationAndSave(s, imageSize, cameraMatrix, distCoeffs,
+                                  imagePoints))
+            mode = CALIBRATED;
+        else
+            mode = DETECTION;
+    }
+
+    if (view.empty())
+    {
+        ROS_DEBUG_STREAM("Empty view");
+        if (mode != CALIBRATED && !imagePoints.empty())
+        {
+            runCalibrationAndSave(s, imageSize, cameraMatrix, distCoeffs,
+                                  imagePoints);
+        }
+        calibFinished = true;
+        ROS_DEBUG_STREAM("Exiting after empty view. Mode: " << mode);
+        ros::shutdown();
+        return;
+    }
+
+    imageSize = view.size();
+    if (s.flipVertical) flip(view, view, 0);
+
+    vector<Point2f> pointBuf;
+
+    bool found;
+
+    int chessBoardFlags = CALIB_CB_ADAPTIVE_THRESH | CALIB_CB_NORMALIZE_IMAGE;
+
+    if (!s.useFisheye)
+    {
+        chessBoardFlags |= CALIB_CB_FAST_CHECK;
```

Comment:
You could probably just combine all of these flags into an in-line ternary in the actual function call, or externally and keep it all on line 58. It would help to compact this a little bit.